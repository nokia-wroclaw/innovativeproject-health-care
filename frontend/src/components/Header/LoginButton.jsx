import React from "react";
import { Button, Popup } from "semantic-ui-react";
import LoginForm from "./LoginForm";
import colors from "../../styles/colors";

const LoginButton = () => {
  return (
    <Popup
      trigger={
        <Button
          content="login"
          inverted
          color={colors.headerLoginButton}
          floated="right"
        />
      }
      content={<LoginForm />}
      on="click"
      position="bottom right"
    />
  );
};

export default LoginButton;
