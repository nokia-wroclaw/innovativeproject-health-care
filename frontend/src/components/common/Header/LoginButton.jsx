import React from "react";
import { Button, Popup } from "semantic-ui-react";
import LoginForm from "../LoginForm";
import * as colors from "../../../styles/colors";

const LoginButton = () => {
  return (
    <Popup
      trigger={
        <Button
          content="Log in"
          inverted
          color={colors.headerLoginButton}
          floated="right"
        />
      }
      content={<LoginForm />}
      on="click"
      position="bottom right"
    />
  );
};

export default LoginButton;
