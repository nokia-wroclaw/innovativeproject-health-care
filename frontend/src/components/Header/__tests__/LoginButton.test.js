import React from "react";
import { shallow } from "enzyme";
import LoginButton from "../LoginButton";

describe("LoginButton", () => {
  it("should match snapshot", () => {
    expect(shallow(<LoginButton />)).toMatchSnapshot();
  });
});
