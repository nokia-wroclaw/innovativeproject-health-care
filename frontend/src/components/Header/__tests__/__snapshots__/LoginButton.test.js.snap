// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`LoginButton should match snapshot 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <LoginButton />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "class",
    "props": Object {
      "content": <Connect(LoginForm) />,
      "disabled": false,
      "keepInViewPort": true,
      "on": "click",
      "position": "bottom right",
      "trigger": <Button
        as="button"
        color="grey"
        content="Login"
        floated="right"
        inverted={true}
      />,
    },
    "ref": null,
    "rendered": null,
    "type": [Function],
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "class",
      "props": Object {
        "content": <Connect(LoginForm) />,
        "disabled": false,
        "keepInViewPort": true,
        "on": "click",
        "position": "bottom right",
        "trigger": <Button
          as="button"
          color="grey"
          content="Login"
          floated="right"
          inverted={true}
        />,
      },
      "ref": null,
      "rendered": null,
      "type": [Function],
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
